src/evaluators/gather/PHAL_GatherScalarNodalParameter_Def.hpp

```diff
@@ -70,23 +70,29 @@ evaluateFields(typename Traits::EvalData workset)
 
   // Distributed parameter vector
   const auto p      = workset.distParamLib->get(this->param_name);
-  const auto p_data = Albany::getLocalData(p->overlapped_vector().getConst());
+  const auto p_data = Albany::getDeviceData(p->overlapped_vector().getConst());
 
   // Parameter dof numbering info
-  const auto p_elem_dof_lids = p->get_dof_mgr()->elem_dof_lids().host();
+  const auto p_elem_dof_lids = p->get_dof_mgr()->elem_dof_lids().dev();
 
   // Mesh elements
   const auto ws = workset.wsIndex;
-  const auto elem_lids = workset.disc->getElementLIDs_host(ws);
+  const auto elem_lids  = workset.disc->getWsElementLIDs();
+  const auto elem_lids_dev = Kokkos::subview(elem_lids.dev(),ws,ALL);
 
-  for (std::size_t cell = 0; cell<workset.numCells; ++cell) {
-    const auto elem_LID = elem_lids(cell);
+  const int nnodes = this->numNodes;
+
+  auto& val = this->val;
```

Comment:
Does the ref work here? A ref is like a pointer, so I expect `KOKKOS_LAMBDA` to not work correctly on CUDA... I would just use a copy here, since it's a shallow copy anyways.

Reply:
So this worked on Perlmutter and Weaver without warning but I'm not against changing this. Should it just be `const auto` instead?

Reply:
Actually, I think you can use `KOKKOS_CLASS_LAMBDA` and you won't have to do these copies at all. Just `auto` is probably best for a view but I think `const auto` is okay too in this case because you're not modifying the view, just the values in the view.

Reply:
Yes, we can use KOKKOS_CLASS_LAMBDA, now that C++17 is mandatory in Trilinos.

Reply:
Okay, I'll use KOKKOS_CLASS_LAMBDA. I think that should let me clean up some of the other variables that are just pointing to class members too.